lib/puppet/provider/foreman_hostgroup/rest_v2.rb

```diff
@@ -0,0 +1,316 @@
+Puppet::Type.type(:foreman_hostgroup).provide(:rest_v2) do
+
+  confine :feature => :apipie_bindings
+
+  # when both rest and rest_v2 providers are installed, use this one
+  def self.specificity
+    super + 1
+  end
+
+  def settings
```

Comment:
It doesn't have to be done in this PR, but we should keep this part consistent for all apipie resources. Currently that's only the smartproxy, so shouldn't be much work. I do wonder if we can abstract this to some helper class.